src/types/pointer.md

```diff
@@ -81,19 +81,60 @@ r[type.pointer.smart]
 
 The standard library contains additional 'smart pointer' types beyond references and raw pointers.
 
-## Bit validity
+## Pointer values and representation
 
-r[type.pointer.validity]
+r[type.pointer.value]
 
-r[type.pointer.validity.pointer-fragment]
-Despite pointers and references being similar to `usize`s in the machine code emitted on most platforms,
-the semantics of transmuting a reference or pointer type to a non-pointer type is currently undecided.
-Thus, it may not be valid to transmute a pointer or reference type, `P`, to a `[u8; size_of::<P>()]`.
+r[type.pointer.value.thin]
+Each thin pointer consists of an address and an optional [provenance][type.pointer.provenance]. The address refers to which byte the pointer points to. The provenance refers to which bytes the pointer is allowed to access, and the allocation those bytes are within.
 
-r[type.pointer.validity.raw]
-For thin raw pointers (i.e., for `P = *const T` or `P = *mut T` for `T: Sized`),
-the inverse direction (transmuting from an integer or array of integers to `P`) is always valid.
-However, the pointer produced via such a transmutation may not be dereferenced (not even if `T` has size zero).
+> [!NOTE]
+> A pointer that does not have a provenance may be called an invalid or dangling pointer.
```

Comment:
*Dangling pointer* is used differently elsewhere in the reference: [link to the `undefined.dangling` section](https://doc.rust-lang.org/reference/behavior-considered-undefined.html#dangling-pointers).